### ISSUE TITLE
Requirements do not properly install

### ISSUE THREAD

**Issue by mikeyshulman** [plasticityai/magnitude#21]
clean install in a virtual environment does not install requirements. 
```
pip 18.0    
Python 3.6.3    

➜  virtualenv -p python3 foo; source foo/bin/activate; pip3 install pymagnitude                                         
In [1]: from pymagnitude import Magnitude                           
---------------------------------------------------------------------------                                                             
ModuleNotFoundError                       Traceback (most recent call last)                                                             
<ipython-input-1-a4fc6d35defa> in <module>()                        
----> 1 from pymagnitude import Magnitude                           

/tmp/foo/lib/python3.6/site-packages/pymagnitude/__init__.py in <module>()                                                              
     11 import hashlib                                              
     12 import heapq                                                
---> 13 import lz4.frame                                            
     14 import math                                                 
     15 import operator                                             

ModuleNotFoundError: No module named 'lz4'                          
```

**Comment by AjayP13** [comment]
Hi,

Thanks for reporting. I just tested it in a fresh docker container ('python:latest' which uses Python 3.6.5 and pip 18.0. I created a virtualenv, activated it, and installed it and it seemed to work fine. 

This might be a issue with your particular environment. The installation does require compiling some native code for its dependencies which might be where things are going wrong. As a note, we try to avoid this whenever possible. When I built the installation, I created eggs (which contain pre-compiled objects for all of the native code) for all the major platforms (32/64-bit macOS, Linux, and 64-bit Windows) for `pymagnitude`. Unfortunately, Python eggs don't recursively pre-compile dependencies of the main project (like `lz4`) that have native code. So I also created a custom install step that creates eggs for all the dependencies so that no compiler would be necessary on all the major platforms.

When any egg installation fails or you are not on a major platform (32/64-bit macOS, Linux, and 64-bit Windows), it falls back to trying to compile the dependencies on your local computer using whatever C/C++ compiler you have. If you don't have a C/C++ compiler available on your machine it can cause issues.

The following information would be helpful to debug:

**OS: ** ??
**OS Version: ** ??
**CPU: ** (Intel or ARM? 32-bit or 64-bit?)

And please re-run the installation in a new, fresh virtualenv with the command `pip3 install pymagnitude -vvvv` and paste the resulting log.

**Comment by mikeyshulman** [comment]
```
Distributor ID: Ubuntu            
Description:    Ubuntu 17.10      
Release:        17.10             
Codename:       artful
```

I started looking at `setup.py`, but the logic was a little hard to follow. If it helps, forcing pip to use `--no-binary :all:` fixes the problem, which means that somehow `requirements.txt` is maybe not getting picked up?

**Comment by AjayP13** [comment]
Hm, glad you got it working. Can you run the failing install with the command `pip3 install pymagnitude -vvvv` and paste the resulting log here. That should print the output of setup.py in verbose mode and show what is failing hopefully. Would be helpful to figure this out for future users who might run into this issue.

**Comment by mikeyshulman** [comment]
I can no longer reproduce the issue, even after blowing away my wheel cache. Sorry.
I'm going to close the issue.

**Comment by AFAgarap** [comment]
@AjayP13 The problem persists. I'm trying to install `pymagnitude` from cache in GitHub Actions, and it's failing to install the dependencies.
